be/src/column/json_column.cpp

```diff
@@ -307,17 +348,27 @@ void JsonColumn::append(const Column& src, size_t offset, size_t count) {
     if (other_json->is_flat_json() && !is_flat_json()) {
         // only hit in AggregateIterator (Aggregate mode in storage)
         DCHECK_EQ(0, this->size());
-        init_flat_columns(other_json->_flat_column_paths, other_json->_flat_column_types);
+        std::vector<ColumnPtr> copy;
+        for (const auto& col : other_json->_flat_columns) {
+            copy.emplace_back(col->clone_empty());
+        }
+        set_flat_columns(other_json->flat_column_paths(), other_json->flat_column_types(), copy);
     }
 
     if (is_flat_json()) {
         DCHECK(src.is_object());
-        DCHECK(_flat_column_paths.size() == other_json->_flat_columns.size());
-        DCHECK(_flat_columns.size() == other_json->_flat_column_paths.size());
+        DCHECK_EQ(_flat_column_paths.size(), other_json->_flat_column_paths.size());
+        DCHECK_EQ(_flat_column_paths.size(), other_json->_flat_column_types.size());
+        DCHECK_EQ(_flat_column_types.size(), other_json->_flat_column_paths.size());
 
-        for (size_t i = 0; i < _flat_columns.size(); i++) {
+        DCHECK_EQ(_flat_columns.size(), other_json->_flat_columns.size());
+
+        for (size_t i = 0; i < _flat_column_paths.size(); i++) {
```

Comment:
Is it possiable `this` is flat json. has flat path A,B,C. other is also flat json. but only has A/B, not has C?

Reply:
this PR never, ColumnReader will promise all json schema is same, but next PR maybe, different chunk may different json schema, but it's will promise never call this function